pkg/storage/gcs.go

```diff
@@ -205,3 +230,22 @@ func newGCSStorage(ctx context.Context, gcs *backup.GCS, opts *ExternalStorageOp
 	}
 	return &gcsStorage{gcs: gcs, bucket: bucket}, nil
 }
+
+func hasSSTFiles(ctx context.Context, bucket *storage.BucketHandle, prefix string) bool {
+	it := bucket.Objects(ctx, &storage.Query{Prefix: prefix})
```

Comment:
consider [fetching the `Name` field only](https://pkg.go.dev/cloud.google.com/go/storage#Query.SetAttrSelection):

```suggestion
	query := storage.Query{Prefix: prefix}
	query.SetAttrSelection([]string{"Name"})
	it := bucket.Objects(ctx, &query)
```